runtime/src/main/java/org/corfudb/runtime/collections/TxBuilder.java

```diff
@@ -0,0 +1,270 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.CorfuOptions;
+import org.corfudb.runtime.CorfuStoreMetadata.Timestamp;
+import org.corfudb.runtime.object.transactions.Transaction;
+import org.corfudb.runtime.object.transactions.TransactionType;
+import org.corfudb.runtime.view.ObjectsView;
+import org.corfudb.runtime.view.TableRegistry;
+
+/**
+ * Created by zlokhandwala on 2019-08-05.
+ */
+public class TxBuilder {
+
+    private final ObjectsView objectsView;
+    private final TableRegistry tableRegistry;
+    private final String namespace;
+    private Timestamp timestamp;
+    private final List<Runnable> operations;
+
+    Set<Descriptors.FieldDescriptor.Type> versionTypes = new HashSet<>(Arrays.asList(
+            Descriptors.FieldDescriptor.Type.INT32,
+            Descriptors.FieldDescriptor.Type.INT64,
+            Descriptors.FieldDescriptor.Type.UINT32,
+            Descriptors.FieldDescriptor.Type.UINT64,
+            Descriptors.FieldDescriptor.Type.SFIXED32,
+            Descriptors.FieldDescriptor.Type.SFIXED64
+    ));
+
+    /**
+     * Creates a new TxBuilder.
+     *
+     * @param objectsView   ObjectsView from the Corfu client.
+     * @param tableRegistry Table Registry.
+     * @param namespace     Namespace boundary defined for the transaction.
+     */
+    @Nonnull
+    TxBuilder(@Nonnull final ObjectsView objectsView,
+              @Nonnull final TableRegistry tableRegistry,
+              @Nonnull final String namespace) {
+        this.objectsView = objectsView;
+        this.tableRegistry = tableRegistry;
+        this.namespace = namespace;
+        this.operations = new ArrayList<>();
+    }
+
+    private void validateNamespace(Table table) {
+        if (table.getNamespace().equals(this.namespace)) {
+            return;
+        }
+        throw new IllegalArgumentException(
+                String.format("Transaction namespace: %s. Attempted transaction on table in namespace: %s",
+                        this.namespace, table.getNamespace()));
+    }
+
+    private <M extends Message> void validateVersion(@Nullable M previousMetadata,
+                                                     @Nullable M userMetadata) {
+        if (userMetadata == null) {
```

Comment:
please use Optional instead of checking for null

Reply:
Either make it Optional as Slava said or NonNull.